### PR TITLE
Add a pseudo-random number generator to CTS

### LINKED ISSUE #2881: Add a pseudo-random number generator to CTS
It would be useful to have a pseudo-random number generator for CTS. It could be used for generating deterministic random testcases (e.g. CFGs).  Math.random is insufficient because it can't be seeded at runtime and thus differs compile-to-compile. However, the core function of Math.random is a useful base for the PRNG, generate a random number in the range [0, 1).

### PR DESCRIPTION
**Pull request by zoddicus**
This adds a utility class, PRNG, that provides a seed-able deterministic random number generator, to assist generating
large sets of test cases without having to encode each one into a source file.

It is based off of TinyMT
(https://github.com/MersenneTwister-Lab/TinyMT), a small state version of Mersenne Twister.

Fixes #2881

<hr>

**Requirements for PR author:**

- [x] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [x] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [x] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [x] Tests are properly located in the test tree.
- [x] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [x] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [x] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

File: src/webgpu/util/prng.ts
Comment (kainino0x) [inline src/webgpu/util/prng.ts:1]: Please add this new file to `docs/helper_index.txt`. (I doubt that file is up to date anymore, as I've forgotten about it for a while, but good to do the best we can.)

**Review by kainino0x — APPROVED** [review]
I don't know the algorithm exactly but the test results look great. LGTM!

**Review by alan-baker — APPROVED** [review]
Thanks for this. LGTM

File: src/webgpu/util/prng.ts
Comment (zoddicus) [reply src/webgpu/util/prng.ts:1]: Done

**Pull request merged**